daemon/pod/container.go

```diff
@@ -267,19 +267,18 @@ func (c *Container) Remove() error {
 }
 
 // Container operations:
-
-func (c *Container) attach(stdin io.ReadCloser, stdout io.Writer, rsp chan<- error) error {
+func (c *Container) attach(stdin io.ReadCloser, stdout io.WriteCloser, rsp chan<- error) error {
 	if c.p.sandbox == nil || c.descript == nil {
 		err := fmt.Errorf("container not ready for attach")
 		c.Log(ERROR, err)
 		return err
 	}
 
-	var stderr io.Writer
+	var stderr io.WriteCloser
 	if stdout != nil {
 		if !c.hasTty() {
-			stderr = stdcopy.NewStdWriter(stdout, stdcopy.Stderr)
-			stdout = stdcopy.NewStdWriter(stdout, stdcopy.Stdout)
+			stderr = &writeCloser{stdcopy.NewStdWriter(stdout, stdcopy.Stderr), stdout}
+			stdout = &writeCloser{stdcopy.NewStdWriter(stdout, stdcopy.Stdout), stdout}
```

Comment:
hmm, good point! Impact of a second call to Close() is undefined. We should avoid it IMO.